truffle/contracts/Splitter.sol

```diff
@@ -0,0 +1,106 @@
+pragma solidity 0.5.10;
+
+import './Owned.sol';
+
+contract Splitter is Owned {
+    
+    struct Person {
+        address wallet;
+        bytes32 name;
+        uint    pendingWithdrawals;
+        uint    index;
+        bool    isParticipant;
+    }
+    
+    mapping(address => Person) participantStorage;
+    address[] participantList;
+    uint public balance;
+
+    event LogParticipantAdded(address participant, bytes32 name);
```

Comment:
`msg.sender` and `msg.value` (if payable) are useful for reconstucting the contract state from logs so it's a good habit to start out with those, e.g. `...LogSomething(address sender, uint amount, ...)`